spec/serializers/site_serializer_spec.rb

```diff
@@ -192,6 +192,75 @@
     end
   end
 
+  describe "#anonymous_sidebar_sections" do
+    fab!(:user) { Fabricate(:user) }
+    fab!(:public_sidebar_section) do
+      Fabricate(:sidebar_section, title: "Public section", user: user, public: true)
+    end
+    fab!(:private_sidebar_section) do
+      Fabricate(:sidebar_section, title: "Private section", user: user, public: false)
+    end
+
+    it "is not included in the serialised object when user is not anonymous" do
+      guardian = Guardian.new(user)
+
+      serialized = described_class.new(Site.new(guardian), scope: guardian, root: false).as_json
+    end
+
+    it "includes only public sidebar sections serialised object when user is anonymous" do
+      serialized = described_class.new(Site.new(guardian), scope: guardian, root: false).as_json
+      expect(serialized[:anonymous_sidebar_sections].map(&:title)).to eq(["Public section"])
+    end
+
+    it "eager loads sidebar_urls" do
+      public_section_link =
+        Fabricate(:custom_sidebar_section_link, user: user, sidebar_section: public_sidebar_section)
+      # warmup
+      described_class.new(Site.new(guardian), scope: guardian, root: false).as_json
```

Comment:
There should be no need to warm up here because we're not re-using the same serializer instance.

Reply:
I checked and we need it because SiteSerializer is using fragment_cache